implementation/src/main/java/io/smallrye/config/NameIterator.java

```diff
@@ -0,0 +1,348 @@
+package io.smallrye.config;
+
+import java.util.NoSuchElementException;
+
+import io.smallrye.common.constraint.Assert;
+
+/**
+ * An iterator for property name strings.
+ */
+public final class NameIterator {
+    /**
+     * Configuration key maximum allowed length.
+     */
+    public static final int MAX_LENGTH = 2048;
+    private static final int POS_MASK = 0x0FFF;
+    private static final int POS_BITS = 12;
+    private static final int SE_SHIFT = 32 - POS_BITS;
+
+    private final String name;
+    private int pos;
+
+    public NameIterator(final String name) {
+        this(name, false);
+    }
+
+    public NameIterator(final String name, final boolean startAtEnd) {
+        this(name, startAtEnd ? name.length() : -1);
+    }
+
+    public NameIterator(final String name, final int pos) {
+        Assert.checkNotNullParam("name", name);
+        if (name.length() > MAX_LENGTH)
+            throw new IllegalArgumentException("Name is too long");
+        Assert.checkMinimumParameter("pos", -1, pos);
+        Assert.checkMaximumParameter("pos", name.length(), pos);
+        if (pos != -1 && pos != name.length() && name.charAt(pos) != '.')
+            throw new IllegalArgumentException("Position is not located at a delimiter");
+        this.name = name;
+        this.pos = pos;
+    }
+
+    public void goToEnd() {
+        this.pos = name.length();
+    }
+
+    public void goToStart() {
+        this.pos = -1;
+    }
+
+    /**
+     * Get the cursor position. It will be {@code -1} if the cursor is at the beginning of the string, or {@code name.length()}
+     * if it is at the end.
+     *
+     * @return the cursor position
+     */
+    public int getPosition() {
+        return pos;
+    }
+
+    /*
+     * next-iteration DFA
+     * <any> → <end> ## on EOI
+     * I → <end> ## on '.'
+     * I → Q ## on '"'
+     * Q → I ## on '"'
+     * Q → QBS ## on '\'
+     * QBS → Q ## on any
+     * I → BS ## on '\'
+     * BS → I ## on any
+     */
+    private static final int FS_INITIAL = 0;
+    private static final int FS_QUOTE = 1;
+    private static final int FS_BACKSLASH = 2;
+    private static final int FS_QUOTE_BACKSLASH = 3;
+
+    /*
+     * Iteration cookie format
+     *
+     * Bit: 14...12 11 ... 0
+     * ┌───────┬────────────┐
+     * │ state │ position │
+     * │ │ (signed) │
```

Comment:
I guess this lost formatting somewhere along the way :)

Reply:
I'll fix it :)